### PR TITLE
Fix/icss hashing

### PR DESCRIPTION
**Pull request by axyz**
@sylvesteraswin 

- prevent icss imports to pollute the package name part of the key in the hash cache
- add example of duplicated classes in different atoms

### CODE REVIEW

File: integration-test/packages/atoms/icss-import/index.esm.js.snap [hunk 1]
```diff
@@ -1,5 +1,5 @@
-export var localClass = "n4oS1I";
-export var localComposes = "i6EUck n4oS1I dgII7d";
-export var simpleComposes = "";
```
Comment (axyz) [inline integration-test/packages/atoms/icss-import/index.esm.js.snap:3]: @sylvesteraswin this was definitely wrong before

File: integration-test/packages/atoms/icss-import/index.esm.js.snap [hunk 2]
```diff
@@ -1,5 +1,5 @@
...
-export var complexComposes = "i6EUck n4oS1I dgII7d _1RrCqF pQ96hs";
-export var extend = "_8F86eH"
```
Comment (axyz) [inline integration-test/packages/atoms/icss-import/index.esm.js.snap:5]: also in these some hashes were missing from the imported ones

File: src/postcss/atomic-css-modules/generateScopedName.ts [hunk 3]
```diff
@@ -43,10 +43,9 @@ const generateScopedName: generateScopedName = ({
           const key = `${pkg};${name}`;
           trackClasses.set(key, hash);
         }
-
         // short circuit: electron definition for a given name is always
         // highest priority and unique source of truth.
-        return DEVELOPMENT ? `${name.toUpperCase()}_${hash}` : hash;
```
Comment (axyz) [inline src/postcss/atomic-css-modules/generateScopedName.ts:49]: this return is in the nested function (wakRules) so it doesn't really need a value, but just returning will not override the key later on in this function

File: src/postcss/atomic-css-modules/generateScopedName.ts [hunk 4]
```diff
@@ -58,20 +57,6 @@ const generateScopedName: generateScopedName = ({
         } else {
           pkgName = importedPackage;
         }
-      } else if (/:import/.test(rule.selector)) {
```
Comment (axyz) [inline src/postcss/atomic-css-modules/generateScopedName.ts:61]: @sylvesteraswin this seem to be the culprit: what was happening basically is that classes in a file with a css import were being evaluated twice with one f the time getting the package name of the imported module (e.g. @my-org/simple-class;withSharedVar).

removing it and returning when rule is a ISS import seems to work fine with the hashes and I also ran the integration tests multiple times to check consstency

**Comment by sylvesteraswin** [comment]
👍

**Comment by axyz** [comment]
:+1:

**Pull request merged**